app/services/builders/chartkick/repository_distribution_data.rb

```diff
@@ -34,20 +31,41 @@ def metric
         @metric ||= RepositoryDistributionDataMetrics.const_get(metric_name.to_s.camelize).new
       end
 
+      def intervals
+        @intervals ||= build_distribution_data(retrieve_records)
+      end
+
       def resolve_interval(entity)
         metric.resolve_interval(entity)
       end
 
-      def build_average
-        return if retrieve_records.empty?
+      def total_value
+        @total_value ||= retrieve_records.sum { |record| metric.value_for_average(record) }
+      end
+
+      def total_records
+        @total_records ||= retrieve_records.count
```

Comment:
```suggestion
        @total_records ||= retrieve_records.size
```

so that if records are already in memory it'll save an extra DB query